tests/testthat/test-model-compile.R

```diff
@@ -232,6 +233,55 @@ test_that("dir arg works for cmdstan_model and $compile()", {
   )
 })
 
+test_that("compiling stops on hyphens in stanc_options", {
```

Comment:
This isn't new I just moved it up in the file to keep the compilation tests separate from check_syntax tests